src/lib/operators/insert.cpp

```diff
@@ -41,15 +45,30 @@ class TypedColumnProcessor : public AbstractTypedColumnProcessor {
     DebugAssert(static_cast<bool>(casted_target), "Type mismatch");
     auto& vect = casted_target->values();
 
+    auto target_is_nullable = casted_target->is_nullable();
+
     if (auto casted_source = std::dynamic_pointer_cast<const ValueColumn<T>>(source)) {
       std::copy_n(casted_source->values().begin() + source_start_index, length, vect.begin() + target_start_index);
+      if (casted_source->is_nullable()) {
+        // Values to insert contain null, copy them
+        Assert(target_is_nullable, "Cannot insert NULL into NOT NULL target");
+        std::copy_n(casted_source->null_values().begin() + source_start_index, length,
+                    casted_target->null_values().begin() + target_start_index);
+      }
```

Comment:
Looks like this does not handle float:

<pre>
(debug)> load src/test/tables/int_float_with_null.tbl x
Loading src/test/tables/int_float_with_null.tbl into table "x" ...
(debug)> INSERT INTO x VALUES (1, 2.0)
===
0 rows total (PARSE: 0.000005 ms, COMPILE: 0.000469 ms, EXECUTE: 0.000519 ms (wall time))
(debug)> INSERT INTO x VALUES (NULL, 2.0)
===
0 rows total (PARSE: 0.000004 ms, COMPILE: 0.000195 ms, EXECUTE: 0.000373 ms (wall time))
(debug)> INSERT INTO x VALUES (1, NULL)
[PERF] operator[] used at src/lib/storage/value_column.cpp:37
	Performance can be affected. This warning is only shown once.
<strong>Exception thrown while executing query plan:
  bad lexical cast: source type value could not be interpreted as target</strong>
terminate called after throwing an instance of 'std::logic_error'
  what():  Operator needs to have state Failed or Executed in order to be rolled back.
Abort trap: 6
</pre>